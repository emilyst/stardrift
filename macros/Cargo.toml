[package]
name = "macros"
edition = "2024"
license = "CC0-1.0"
publish = false
description = "Procedural macros for Stardrift configuration"

[lib]
proc-macro = true

[dependencies]
syn = { version = "2.0", features = ["derive", "parsing"] }
quote = "1.0"
proc-macro2 = "1.0"

[package.metadata.release]
release = false

[dev-dependencies]
serde = { version = "1.0", features = ["derive"] }
serde_json = "1.0"
trybuild = "1.0"
