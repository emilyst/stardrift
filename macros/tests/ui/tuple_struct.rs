// This test should fail to compile because tuple structs are not supported
use macros::ConfigDefaults;

#[derive(ConfigDefaults)]
struct TupleStruct(i32, String);

fn main() {}
