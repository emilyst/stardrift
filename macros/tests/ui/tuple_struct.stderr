error: ConfigDefaults only supports structs with named fields
 --> tests/ui/tuple_struct.rs:5:1
  |
5 | struct TupleStruct(i32, String);
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
